### ISSUE TITLE
Legacy Update fails to install on Windows 8.0

### ISSUE THREAD

**Issue by win98se** [kirb/LegacyUpdate#128]
Similar to #104, I got an error (same error code, -2145124329) with KB2937636 on Windows 8.0, then it proceeds to “Installation failed”.

![image](https://user-images.githubusercontent.com/11819816/221263562-cf4390bf-e13c-41b9-bbaa-a18e20295ff6.png)

According to [this document](https://github.com/MicrosoftDocs/SupportArticles-docs/blob/main/support/windows-client/deployment/update-windows-update-agent.md#stand-alone-packages-for-windows-8-and-windows-server-2012), it is the Windows Update Agent update for Windows 8.0 and Windows Server 2012.

My `wuaueng.dll` has a version of 7.8.9200.23250, even newer than 7.8.9200.16693 as mentioned in the above link.

![image](https://user-images.githubusercontent.com/11819816/221264085-543108a3-b22f-4c8c-a881-66ae111baa1c.png)

Hence, I am sure that I don't even need this update at all, because KB2937636 may have lower version than the one I've installed.

Would you add a checking mechanism or unset the Servicing Stack update as mandatory?

**Comment by renodr** [comment]
This is really interesting. Can you search for Package_for_KB2937636_RTM in C:\Windows\servicing?

It's possible that the Windows Update Agent update has been superseded. For example, https://support.microsoft.com/en-us/topic/december-2014-update-rollup-for-windows-rt-windows-8-and-windows-server-2012-f379ce82-ae16-83e9-601f-867c9aa95264 (KB3013767) seems to replace wuaueng.dll with a version of 17185.

I'll keep experimenting here though

**Comment by renodr** [comment]
Sorry that should be C:\Windows\servicing\Packages

**Comment by win98se** [comment]
Thank you in advance.

There are no files starting with `Package_for_KB2937636_RTM` in `C:\Windows\servicing\Packages` here.

**Comment by renodr** [comment]
Hi win98se,

Are you using any custom installation media? In my case, I'm testing with retail Windows 8 Pro media

Can you get us a directory listing of C:\Windows\servicing\Packages on your system? I'd like to see exactly which KBs that Windows thinks it has installed so I can see if I can figure out what is providing that 23250 version

**Comment by win98se** [comment]
Apologies for late reply; I used an ISO made by someone else that includes almost all updates until December 2021.

Attached the directory listing of my `C:\Windows\servicing\Packages` - [Windows 8 with updates (2021-12).txt](https://github.com/kirb/LegacyUpdate/files/11286613/Windows.8.with.updates.2021-12.txt)

**Comment by renodr** [comment]
> Apologies for late reply; I used an ISO made by someone else that includes almost all updates until December 2021.
> 
> Attached the directory listing of my `C:\Windows\servicing\Packages` - [Windows 8 with updates (2021-12).txt](https://github.com/kirb/LegacyUpdate/files/11286613/Windows.8.with.updates.2021-12.txt)

Thank you for this file, it proved extremely useful in figuring out what was going on!

I made a couple of interesting discoveries. I did a fresh install of Windows 8 Pro, but in 32-bit. Most of the updates that you have there are from 2021 and 2022 (which makes sense!) - but they're for Windows Server 2012, at least if you check Microsoft's website. However, it turns out that you can feed Windows 8 updates for Windows 8 Embedded, and it won't care.

I did a bit of digging through the file and searched a lot of your updates, until I found one update that clued me in - KB4598297. It's a Security Only update... which updates the Windows Update Engine! I can confirm that the version number is in fact 23250. As a prerequisite though, you need a newer SSU than what Windows 8's retail installer will give you. The version that [KB4598297](https://support.microsoft.com/en-us/topic/january-12-2021-kb4598297-security-only-update-3dd1c96b-cd9f-0978-af71-908711f3baec) references is from July 14th of 2020, and goes to [KB4566426](https://support.microsoft.com/en-us/topic/servicing-stack-update-for-windows-server-2012-july-14-2020-2a89df81-5041-90d2-0111-e47c96c286e2). You didn't have that update installed though, because it's been superseded by another one - [KB5011571](https://support.microsoft.com/en-us/topic/kb5011571-servicing-stack-update-for-windows-server-2012-march-8-2022-49809c9f-1a2f-40d1-9ca3-791d820190e4)

I've confirmed that KB4598297 and KB4566426 installed correctly, but haven't tried KB5011571.

My next approach is going to be to install KB5011571 and KB4598297 and then run through a complete installation of all Windows Updates offered through Windows Update.

If everything is stable after that, I'll make a PR to install these both through LU in substitution of the WUA for Windows 8.0, which is no longer needed after that is done... and also fixes [Microsoft ADV990001](https://msrc.microsoft.com/update-guide/vulnerability/ADV990001) - which was never officially fixed for Windows 8.0.

**Comment by renodr** [comment]
It turns out that KB5011571 and KB4566426 are not required for KB4598297 to install. I'm waiting on a test run of Windows Update to verify that everything is good and then will submit a PR.

**Comment by renodr** [comment]
PR submitted as #145
